src/openslide-vendor-generic-tiff.c

```diff
@@ -245,6 +259,14 @@ static bool generic_tiff_open(openslide_t *osr,
   // sort tiled levels
   g_ptr_array_sort(level_array, width_compare);
 
+  // get icc profile size, if present
+  uint32_t icc_profile_size;
+  void *icc_profile;
+  if (TIFFGetField(ct.tiff, TIFFTAG_ICCPROFILE,
```

Comment:
We're currently on the last TIFF directory, when we probably want the highest-resolution one.